projects/igniteui-angular/src/lib/grids/grid/grid-api.service.ts

```diff
@@ -43,12 +43,16 @@ export class IgxGridAPIService extends GridBaseAPIService<IgxGridComponent> {
             names.forEach((colName) => {
                 const grExprIndex = groupingState.findIndex((exp) => exp.fieldName === colName);
                 const grpExpandState = this.grid.groupingExpansionState;
-                /* remove expansion states related to the cleared group
-                and all with deeper hierarchy than the cleared group */
-                this.grid.groupingExpansionState = grpExpandState
-                    .filter((val) => {
-                        return val.hierarchy && val.hierarchy.length <= grExprIndex;
-                    });
+                /* Do not set the new instance produced by filter
+                when there are not any expansion states */
+                if (grpExpandState.length) {
```

Comment:
As discussed today I think the correct logic would be:

```typescript
var newState = this.grid.groupingExpansionState = grpExpandState
    .filter((val) => {
         return val.hierarchy && val.hierarchy.length <= grExprIndex;
    });
if (newState.length !==this.grid.groupingExpansionState.length) {
    this.grid.groupingExpansionState = newState;
}
```

Reply:
@ChronosSF You are right as this would also prevent the event from emitting when a different column than the one with expansion state is unpinned so it is a better solution.

I noticed that the first line of the suggested code has a typo:
var newState = **this.grid.groupingExpansionState** = grpExpandState
    .filter((val) => { ...

Should be just:
 var newState = grpExpandState
                .filter((val) => {...